pages/architecture/distributed_registry.rst

```diff
@@ -0,0 +1,230 @@
+    #. The node information is published so that every (other) node in the network knows about the vendor's node.
+#. Vendor registration:
+    #. The vendor generates a CSR (for the vendor's CA certificate) and submits it to Nuts Registry.
+       TODO: Nuts Registry shouldn't be responsible for issuing certificates, since we're also issuing them
+       from Nuts Discovery? Or maybe they're RA's in front of the Nuts Foundation CA?
+    #. A Nuts administrator accepts the CSR and issues the vendor CA certificate.
+    #. The vendor creates the registration event and signs it using the issued CA certificate.
+
+VendorRegistration Event
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+======================  =====
+Field                   Format
+======================  =====
+Vendor CA certificate   PEM encoded X.509
+======================  =====
+
+Validation
+""""""""""
+
+#. Is the vendor CA certificate signed by the Nuts Foundation?
+
+
+Registering a contract (vendor - care organisation)
+---------------------------------------------------
+
+Through this process a vendor registers a care organisation as its client.
+Afterwards, the vendor can register endpoints for the care organisation it serves through its node.
+
+Steps:
+
+#. The vendor generates a CSR for the vendor's contract with the care organisation and signs it
+   using its vendor CA certificate (see the table below for the prescribed fields of the certificate).
+#. The vendor creates the contract proposal and signs it using the (just issued) contract certificate.
+#. The vendor creates the contract approval and signs it (again) using the contract certificate (see note below).
+
+Contract certificate fields:
+
+====================  =====
+Field                 Value
+====================  =====
+Validity              3 years
+Key usage             Digital signature
+CA?                   false
+Issuer                Vendor CA certificate
+Subject.CN            Care organisation's name or unit within the organisation
+Subject.O             Care organisation's name
+Subject.ST            Province in which the care organisation is located
+Subject.C             Care organisation's country
+Subject.serialNumber  Chamber of Commerce registration number of the care organisation
+====================  =====
+
+ContractProposal Event
+^^^^^^^^^^^^^^^^^^^^^^
+
+This event is emitted (and thus signed) by the vendor and proposes the contract to the care organisation.
+
+======================  =====
+Field                   Format
+======================  =====
+Contract ID             UUID type 4
+Care organisation       OID
+Vendor                  OID
+Start date              RFC 3339 timestamp
+End date                RFC 3339 timestamp (optional)
+Certificate             PEM encoded X.509
+======================  =====
+
+Validation
+""""""""""
+
+#. Is this event referring to an event stream of the same vendor?
```

Comment:
event stream doesn't need to be vendor specific? What does it add?

Reply:
Agree. Was a case of premature optimisation.